nibabel/parrec.py

```diff
@@ -725,19 +734,26 @@ def get_bvals_bvecs(self):
         n_slices, n_vols = self.get_data_shape()[-2:]
         bvals = self.image_defs['diffusion_b_factor'][reorder].reshape(
             (n_slices, n_vols), order='F')
-        # All bvals within volume should be the same
-        assert not np.any(np.diff(bvals, axis=0))
+        if not self.strict_sort:
```

Comment:
Can you think of a better place for these checks?  Maybe there's a way to check these after doing the sorting?